### PR TITLE
Fix #939: Add Issue and Pull Request Template

### LINKED ISSUE #939: Issue and Pull Request Templates
It would be super useful to have an Issue and a Pull Request template as described [here](https://github.com/blog/2111-issue-and-pull-request-templates).

This would allow better management of submitted issues and PRs by the maintainers by maintaining consistency from contributors.

### Plan
- [ ] Discuss the correct format for the various templates.
- [ ] Create a `.github` directory.
- [ ] Create the template files and add them to the repository.

### PR DESCRIPTION
**Pull request by deshraj**
Fixes #939

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+_Thanks for taking the time to report a bug in the EvalAI project, or propose a new feature! Before filing a new issue, please do a quick search to check that it hasn't already been filed on the [issue tracker](https://github.com/Cloud-CV/EvalAI/issues)._
```
Comment (TroJan) [inline .github/ISSUE_TEMPLATE.md]: Can you make if `Before creating`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+IMPORTANT NOTES (please read, then delete):
+
+* The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: introduce the first version of the collection editor.".
```
Comment (TroJan) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we do introduce --> Introduce ?

File: README.md [hunk 3]
```diff
@@ -92,7 +92,7 @@ Follow this guide to setup your development machine.
 
 ## Contribution guidelines
 
-If you are interested in contributing to EvalAI, follow your [contribution guidelines](https://github.com/Cloud-CV/EvalAI/blob/master/CONTRIBUTING.md).
+If you are interested in contributing to EvalAI, follow your [contribution guidelines](https://github.com/Cloud-CV/EvalAI/blob/master/.github/CONTRIBUTING.md).
```
Comment (TroJan) [inline README.md]: Can we do your --> our?

**Comment by deshraj** [comment]
@varunagrawal please let us know if you have thoughts about the templates. Would be happy to hear your suggestions. :)

**Comment by deshraj** [comment]
@varunagrawal: I am merging this for now. Let us know if you have thoughts. We can iterate over it. :)

**Comment by varunagrawal** [comment]
Looks great. I would have preferred having a separate document that provides info on how to file bugs or submit PRs and the templates just being placeholders for people to fill in. But as you said, we can think about this further and revisit. :)

**Comment by deshraj** [comment]
Cool. Please feel free to create pull request. :)

**Pull request merged**